packages/@expo/metro-config/src/transform-worker/dot-env-development.ts

```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright 2025-present 650 Industries (Expo). All rights reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import * as dotenv from 'dotenv';
+import { expand as dotenvExpand } from 'dotenv-expand';
+
+export function parseEnvFile(src: string, isClient: boolean): Record<string, string> {
+  const expandedEnv: Record<string, string> = {};
+  const envFileParsed = dotenv.parse(src);
+
+  if (envFileParsed) {
+    const allExpandedEnv = dotenvExpand({
+      parsed: envFileParsed,
+      processEnv: {},
```

Comment:
I think this part may cause some issues here. Without passing the "current environment variables", we can't do things like `EXPO_PUBLIC_SOME=value npx expo start`. Note that `dotenv-expand` also mutates this object, so I think passing a clone of `process.env` is most suitable here.

Keep in mind that in `@expo/env` we did variable expansion as a final step, not as intermediate in-between steps. I forgot the exact reason, but there might be some differences in behavior.

Reply:
We still use the `process.env` logic from before, it's added to the serializer. We just apply the dotenv changes over those changes.